lib/lf/geometry/segment_o1.h

```diff
@@ -21,6 +24,20 @@ class SegmentO1 : public Geometry {
       const Eigen::MatrixXd& local) const override;
   std::unique_ptr<Geometry> SubGeometry(dim_t codim, dim_t i) const override;
 
+  /** @brief creation of child geometry for the sake of mesh refinement 
+   *
+   * @see Geometry::ChildGeometry()
+   * @see RefinementPattern
+   * @param ref_pattern three refinement patterns are supported
+   * - rp_copy: just copies the geometry information of the segment
+   * - rp_split, rp_regular: split edge in the middle.
+   * @param anchor not used
+   * @param selector = 0 for splitting refinement pattern returns half edge
+   * adjacent to endpoint 0, =1 returns the other half.
+   */
+  virtual std::vector<std::unique_ptr<Geometry>>
+  ChildGeometry(const RefinementPattern &ref_pattern) const override;
```

Comment:
In Principle there is no need to document this method also because it overrides a method from the interface (which is documented). Doxygen will then just copy that documentation over to this method.